packages/x-components/src/x-modules/history-queries/store/__tests__/actions.spec.ts

```diff
@@ -238,4 +239,35 @@ describe('testing history queries module actions', () => {
       expect(store.state.historyQueries).toEqual(historyQueries);
     });
   });
+
+  describe('toggleHistoryQueries', () => {
+    beforeEach(() => {
+      const historyQueries: HistoryQuery[] = createHistoryQueries('gato', 'perro');
+      resetStateWith({ historyQueries, isEnabled: true });
+    });
+
+    it("doesn't allow new history queries to be set on disabled", async () => {
+      await store.dispatch('toggleHistoryQueries', false);
+
+      const newHistoryQuery = createHistoryQuery({ query: 'caballo' });
+      await store.dispatch('setHistoryQueries', [newHistoryQuery]);
+
+      expectHistoryQueriesToEqual([]);
+    });
+
+    it('deletes the history queries when disabled', async () => {
+      const historyQueries: HistoryQuery[] = createHistoryQueries('gato', 'perro');
+      expectHistoryQueriesToEqual(historyQueries);
+
+      await store.dispatch('toggleHistoryQueries', false);
+      expectHistoryQueriesToEqual([]);
+    });
+
+    it('stores the enabled/disabled value in local storage', async () => {
+      expect(localStorageService.getItem<boolean>(HISTORY_QUERIES_ENABLED_KEY)).toBe(true);
+
+      await store.dispatch('toggleHistoryQueries', false);
+      expect(localStorageService.getItem<boolean>(HISTORY_QUERIES_ENABLED_KEY)).toBe(false);
```

Comment:
```suggestion
      expect(localStorageService.getItem<boolean>(HISTORY_QUERIES_ENABLED_KEY)).toBe(false);
      await store.dispatch('toggleHistoryQueries', true);
      expect(localStorageService.getItem<boolean>(HISTORY_QUERIES_ENABLED_KEY)).toBe(true);
```

Reply:
This doesn't work. The first expect has to be null because there is nothing stored in the `localStorage`. I'm putting your changes but with `null` instead of `false`.

And maybe add another expect.